src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -615,7 +631,7 @@ private AbstractFrame getClassFrameWhereViolationIsFound(DetailAST ast) {
         final DetailAST prevSibling = ast.getPreviousSibling();
         if (variableDeclarationFrameType == FrameType.CLASS_FRAME
                 && !validateOnlyOverlapping
-                && prevSibling == null
+                && (prevSibling == null || EXPR_TOKENS.contains(ast.getParent().getType()))
```

Comment:
Curious, what changes occur when you remove `prevSibling == null`? Why don't we just look at all expressions?

Reply:
it breaks cases like
```
public int doSideEffect() {
       return bar;
}
```

Reply:
When you say it breaks cases like that, do you mean removing it from the old code? The new code makes sense since you are checking for specific tokens, but why can't we do all expressions?

Reply:
this check (prevSibling == null) relates not only to expressions.

Reply:
> it breaks cases like
> return bar;

Bar is still part of an expression, so I don't see how removing `prevSibling == null` breaks it.

````
    |  `--SLIST -> { [2:26]
    |      |--LITERAL_RETURN -> return [3:7]
    |      |  |--EXPR -> EXPR [3:14]
    |      |  |  `--IDENT -> bar [3:14]
    |      |  `--SEMI -> ; [3:17]
    |      `--RCURLY -> } [4:0]
````

Reply:
I removed the entire line, `&& (prevSibling == null || EXPR_TOKENS.contains(ast.getParent().getType()))`, and false positives came up like:
> `this.booleanField = true;`
> `if (this.ex != null) {`
> `Object plus = "" + this.id;`

It looks like it is requesting everything have a double `this`. It seems to me this line just needs to check there isn't already a `this` but it would have to be examined more.

Reply:
looks like a good idea, I didn't notice it. Will try it later

Reply:
your suggestion worked out well, thanks. I corrected code.